pyccel/codegen/printing/fcode.py

```diff
@@ -2311,7 +2311,7 @@ def _print_Allocate(self, expr):
             elif expr.alloc_type == 'reserve':
                 var_code = self._print(expr.variable)
                 size_code = self._print(expr.shape[0])
-                return f'call {var_code} % reserve({size_code})\n'
+                return f'call {var_code} % clear()\n' + f'call {var_code} % reserve({size_code})\n'
```

Comment:
The `+` seems unnecessary but I guess you introduced it for readability. What about:
```suggestion
                return (f'call {var_code} % clear()\n'
                        f'call {var_code} % reserve({size_code})\n')
```
?